### PR TITLE
improves multi-tx UX: wait for tx confirmation before requesting the next

### PR DESCRIPTION
**Pull request by georgeweiler**

### CODE REVIEW

**Comment by georgeweiler** [comment]
This is a pretty repetitive addition but it works. I will likely push up a refactor soon for improving reusability. Happy to do that later if you need this update merged in ASAP  @CarlBeek

File: src/pages/Transactions/transactionUtils.tsx [hunk 1]
```diff
@@ -98,3 +98,86 @@ export const handleTransaction = async (
...
+  const contract = new web3.Contract(contractAbi, CONTRACT_ADDRESS);
+
+  const transactionParameters: SendOptions = {
+    gasPrice: '0x0055e72a000', // TODO estimate gas value
+    from: account as string,
+    value: TX_VALUE,
+  };
+
+  if (recursionIdx === depositFiles.length) {
+    // stop calling yourself
```
[the remark this answers is not in the document]
Comment (georgeweiler) [reply src/pages/Transactions/transactionUtils.tsx:60]: updated.

**Comment by georgeweiler** [comment]
Pushed a small update to improve code quality

File: src/pages/Transactions/transactionUtils.tsx [hunk 2]
```diff
@@ -31,70 +29,92 @@ const isUserRejectionError = (error: any) => {
...
+      }
+    )
+    .on('error', (error: any) => {
+      handleMultipleTransactions(
+        depositFiles,
+        connector,
+        account,
+        updateTransactionStatus,
+        recursionIdx + 1
+      );
```
Comment (CarlBeek) [inline src/pages/Transactions/transactionUtils.tsx:113]:
```suggestion
```

If the user rejects a TX or something goes wrong, we should not continue presenting more TXs. Instead the user can fix the problem (or not) and continue by pressing the "Make N Deposit transactions" at the top of the page

**Comment by CarlBeek** [comment]
Great work, @georgeweiler!

To clarify what is being done here, currently each deposit launches an instance of the web3 signer (eg Metamask) simultaneously for each deposit. The result is one window per deposit which, if you are making many deposits, results in an unresponsive PC (or even double deposits due to race conditions between the windows). This PR tackles this by iterating through the deposits and only starting a new one once the previous has been signed.

**Review by CarlBeek — CHANGES_REQUESTED** [review]
One problem I've encountered since this update is that the window with the keys does not fill the width of the page anymore. Can I ask you to take a crack at it @georgeweiler?

Screenshot 2020-09-01 at 11 38 33 AM

**Comment by georgeweiler** [comment]
hey @CarlBeek I just pushed a change to solve the loop/rejection issue and the css bug that I was only experiencing on firefox.

**Review by CarlBeek — APPROVED** [review]

**Pull request merged**